sensor/common/registry/registry_store.go

```diff
@@ -75,21 +77,21 @@ type Store struct {
 	// centralRegistryIntegration holds registry integrations sync'd from Central.
 	centralRegistryIntegrations registries.Set
 
-	tlsCheckCache *tlsCheckCacheImpl
+	tlsCheckCache tlscheckcache.Cache
 }
 
-// CheckTLS defines a function which checks if the given address is using TLS.
-// An example implementation of this is tlscheck.CheckTLS.
-type CheckTLS func(ctx context.Context, origAddr string) (bool, error)
-
 // NewRegistryStore creates a new registry store.
 // The passed-in CheckTLS is used to check if a registry uses TLS.
 // If checkTLS is nil, tlscheck.CheckTLS is used by default.
-func NewRegistryStore(checkTLSFunc CheckTLS) *Store {
+func NewRegistryStore(checkTLSFunc tlscheckcache.CheckTLSFunc) *Store {
 	if checkTLSFunc == nil {
 		checkTLSFunc = tlscheck.CheckTLS
```

Comment:
seems like this is the default, right? Perhaps we can remove this and just document it as the default

Reply:
That will break tests / make them try to reach out to the defined hosts.

Reply:
ah ok I think it's because you have the line `tlscheckcache.WithTLSCheckFunc(checkTLSFunc)` below this unconditionally, so if we remove this `nil` check, we set the TLSCheckFunc to `nil` below. Up to you if you would want keep things as-is or just set `tlscheckcache.WithTLSCheckFunc(checkTLSFunc)` conditionally

Reply:
Ahh I see what you mean now. Going to leave it as is to avoid having to create a separate slice of options to make it conditional. Am happy to change it in a follow up as needed.